rows_test.go

```diff
@@ -0,0 +1,586 @@
+			},
+		},
+		{
+			name:  "max int",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "maxint"`,
+			want: []testAtomicTypesRow{
+				{key: "maxint", testString: "hello", testBytes: []byte("hello"), testInt: 9223372036854775807, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:  "min int",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "minint"`,
+			want: []testAtomicTypesRow{
+				{key: "minint", testString: "hello", testBytes: []byte("hello"), testInt: -9223372036854775808, testFloat: 42, testBool: true},
+			},
+		},
+		{
+			name:  "special float positive infinity",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "pinf"`,
+			want: []testAtomicTypesRow{
+				{key: "pinf", testString: "hello", testBytes: []byte("hello"), testInt: 42, testFloat: math.Inf(1), testBool: true},
+			},
+		},
+		{
+			name:  "special float negative infinity",
+			input: `SELECT * FROM TestAtomicTypes WHERE key = "ninf"`,
+			want: []testAtomicTypesRow{
+				{key: "ninf", testString: "hello", testBytes: []byte("hello"), testInt: 42, testFloat: math.Inf(-1), testBool: true},
+			},
+		},
+	}
+
+	// Run tests.
+	for _, tc := range tests {
+
+		rows, err := db.QueryContext(ctx, tc.input)
+		if (err != nil) && (!tc.wantError.query) {
+			t.Errorf("%s: unexpected query error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.wantError.query) {
+			t.Errorf("%s: expected query error but error was %v", tc.name, err)
+		}
+
+		got := []testAtomicTypesRow{}
+		for rows.Next() {
+			var curr testAtomicTypesRow
+			err := rows.Scan(
+				&curr.key, &curr.testString, &curr.testBytes, &curr.testInt, &curr.testFloat, &curr.testBool)
+			if (err != nil) && (!tc.wantError.scan) {
+				t.Errorf("%s: unexpected scan error: %v", tc.name, err)
+			}
+			if (err == nil) && (tc.wantError.scan) {
+				t.Errorf("%s: expected scan error but error was %v", tc.name, err)
+			}
+
+			got = append(got, curr)
+		}
+
+		rows.Close()
+		err = rows.Err()
+		if (err != nil) && (!tc.wantError.close) {
+			t.Errorf("%s: unexpected rows.Err error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.wantError.close) {
+			t.Errorf("%s: expected rows.Err error but error was %v", tc.name, err)
+		}
+
+		if !reflect.DeepEqual(tc.want, got) {
+			t.Errorf("Unexpected rows: %s. want: %v, got: %v", tc.name, tc.want, got)
+		}
+
+	}
+
+	// Drop table.
+	_, err = db.ExecContext(ctx, `DROP TABLE TestAtomicTypes`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestRowsOverflowRead(t *testing.T) {
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	// Set up test table.
+	_, err = db.ExecContext(ctx,
+		`CREATE TABLE TestOverflowRead (
+			key	STRING(1024),
+			testString	STRING(1024),
+			testBytes	BYTES(1024),
+			testInt	INT64,
+			testFloat	FLOAT64,
+			testBool	BOOL
+		) PRIMARY KEY (key)`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = db.ExecContext(ctx,
```

Comment:
Since we are only testing `int64` overflow here, is there anything that the other columns add to this specific test? Do we feel we could just have a table with a `int64` instead?

Reply:
Ah yes this is an older test from when I didn't know go types well/ didn't realise I couldn't really overflow string or byte arrays. in that case I might just move the overflow tests into the type permute function?

Reply:
Sounds good to me

Reply:
Moved